apps/snitch_core/lib/core/domain/order/transitions.ex

```diff
@@ -124,4 +125,38 @@ defmodule Snitch.Domain.Order.Transitions do
   end
 
   def persist_shipment(%Context{valid?: false} = context), do: context
+
+  defp extract_shipping_method_cost(shipping_method_id) do
+    shipping_method = Repo.get(ShippingMethod, shipping_method_id)
```

Comment:
Why involve `Repo`? The package struct already has `shipping_methods`, a list of `shipping_method` embeds, each of them has the cost in it.
I would do this:
```elixir
defp extract_shipping_method_cost(package) do
  sm = Enum.find(package.shipping_methods, fn %{shipping_method_id: id} ->
    id == package.shipping_method_id
  end)
  sm.cost
end

Reply:
Context data should be struct `preloaded` before sending to state machine, if I am not wrong ?

Reply:
You are once again thinking about _triggering the state machine_. Please work on _handling the trigger_ instead.
Moreover, I don't see why you are fetching shipping methods from the DB, **when you already have them, and the costs as well, in the `package`**.